l3backend/l3backend-pdf.dtx

```diff
@@ -156,7 +143,7 @@
 \cs_new_protected:Npn \@@_backend_object_write:nnn #1#2#3
   {
     \@@_backend_object_write_aux:nnn
-      { \@@_backend_object_ref:n {#1} }
+      { \pdf_object_ref:n {#1} }
```

Comment:
Why this change? The `#1` here is already an int value, either value of `c_@@_object_ #1 _int`
```tex
\cs_new_protected:Npn \pdf_object_write:nnn #1#2#3
  {
    \exp_args:Nv \@@_backend_object_write:nnn
      { c_@@_object_ #1 _int } {#2} {#3}
    \bool_gset_true:N \g_@@_init_bool
  }
```
or the object number stored in intarray
```tex
\cs_new_protected:Npn \@@_object_write_indexed:Nnnn #1#2#3#4
  {
    \exp_args:Ne \@@_backend_object_write:nnn
      { \intarray_item:Nn #1 {#2} }
      {#3} {#4}
    \bool_gset_true:N \g_@@_init_bool
  }
```
------

```suggestion
      { \@@_backend_object_ref:n {#1} }
```